### DISCUSSION TITLE
should the project remove copyright header comments from source files?

### DISCUSSION THREAD

**Discussion by obnoxxx** [rook/rook#15000] [General]
The Apache Software foundation (ASF) asks that committers to Apache licensed projects do not add copyright/license headers to source files: https://www.apache.org/legal/apply-license.html#copy-per-file

They are anyways  redundant and don't add value.

I would therefore like to discuss whether the rook project should remove the copyright/license header comments from all source files.

**Comment by obnoxxx** [comment]
@travisn , @BlaineEXE  @jbw976 : wdyt?

**Reply by BlaineEXE** [reply]
I think we should strip out the blocks. I've never particularly enjoyed these.

**Reply by jbw976** [reply]
Actually, the CNCF guidelines mention including them in each file, so that's what I would stick with: https://github.com/cncf/foundation/blob/main/license-notices.md

> Where reasonably possible, each source code and documentation file that is contributed to a CNCF project should contain a copyright notice and license notice.

> There are several reasons to include notices on a file-by-file basis, rather than just in a single LICENSE file in the repo root directory.

**Reply by BlaineEXE** [reply]
Well, I think we have our clear answer on this topic. Thanks, Jared for providing that link.

**Reply by obnoxxx** [reply]
Indeed, thanks, @jbw976  for providing the CNCF  link which my web search had not produced.

I have however found some  projects (non-CNCF) that removed all such comment header blocks in a bulk change to adhere to the ASF preference.

I remember that long ago, we removed all copyright comment headers from files in the Samba project (samba.org), which is a GPL project. We removed them because they were deemed redundant and useless.

But despite @BlaineEXE 's agreement, it seems we cannot remove the headers from the rook project because of the CNCF policy.

**Comment by obnoxxx** [comment]
given CNCF has the policy of having those comment headers, it seems to be safe to close this discussion.
